pkg/storage/local.go

```diff
@@ -52,13 +52,16 @@ func (l *LocalStorage) WalkDir(ctx context.Context, fn func(string, int64) error
 			return nil
 		}
 
-		return fn(f.Name(), f.Size())
+		return fn(path, f.Size())
 	})
 }
 
-// Open a Reader by file name.
-func (l *LocalStorage) Open(ctx context.Context, name string) (ReadSeekCloser, error) {
-	return os.Open(path.Join(l.base, name))
+// Open a Reader by file path.
+func (l *LocalStorage) Open(ctx context.Context, p string) (ReadSeekCloser, error) {
+	if !filepath.IsAbs(p) {
+		p = path.Join(l.base, p)
+	}
+	return os.Open(p)
```

Comment:
this look like a security hole :eyes: why do we support absolute path? (btw use `filepath.Join` instead if `path.Join`.

Reply:
the `path` parameter in `filepath.Walk(, func(path, info))`  is absolute path in max osx, and relative path in linux. So I'm not sure if we should be compatible with both

Reply:
use [`filepath.Rel`](https://golang.org/pkg/path/filepath/#Rel) to force the path to be relative to `l.base`?

Reply:
done